protoc-gen-go.hcl

```diff
@@ -0,0 +1,11 @@
+description = "Go support for Google's protocol buffers."
+binaries = ["protoc-gen-go"]
+test = "protoc-gen-go --version"
+arch = "amd64"
+source = "https://github.com/protocolbuffers/protobuf-go/releases/download/v${version}/protoc-gen-go.v${version}.${os}.${arch}.tar.gz"
```

Comment:
There's no arm64 build so you'll have to replace `${arch}` with `amd64`.

Reply:
not my brightest moment :(
Fixed.

Reply:
actually still broken:
https://github.com/cashapp/hermit-packages/pull/57/checks#step:4:133
for `protoc-gen-go-grpc` ( https://github.com/grpc/grpc-go/releases/download/cmd/protoc-gen-go-grpc/v1.0.1/protoc-gen-go-grpc.v1.0.1.darwin.arm64.tar.gz) which I haven't touched in this pr.
I don't even understand why the test for this is run. Or why it didn't fail my before.
Maybe something to do with the updated CI tests?

Reply:
protoc-gen-go is a prefix of protoc-gen-go-grpc ....

Reply:
@camh- thanks #58 .
I guess should still fix `protoc-gen-go-grpc`

Reply:
maybe this is related to the new hermit release from earlier today?

Reply:
ok, I had a quick scan, I think only jx is the only other package that might have missing arch'es.

Reply:
Aaah yes that would be it for sure.